### DISCUSSION TITLE
What's the recommended size of a state machine?

### DISCUSSION THREAD

**Discussion by capaj** [statelyai/xstate#2554] [Q&A] [answered]
I just joined a project where a whole app is in a single state machine defined in a 5400 LOC yaml file. Is this how xstate should be used or do you recommend multiple smaller state machines? What's the maximum manageable size of the yaml definition?

**Comment by davidkpiano** [comment, marked as answer]
That's a big machine! 😳

If the complexity of the logic warrants such a large machine, so be it. But in general, nested states can become their own machines, and it's usually a good idea to split them up if it becomes too large.

**Reply by pke** [reply]
OT: xstate can parse YAML state charts? That's new to me. How does this work?

**Reply by capaj** [reply]
The project uses yaml, but it's read by a regular yaml parser and used as regular POJO at runtime. I guess authors preferred yaml to POJO.
